tasks/task_manager.hh

```diff
@@ -187,6 +187,19 @@ public:
         // If task has a parent, data concerning its children is updated and sequence number is inherited
         // from a parent and set. Otherwise, it must be set by caller.
         future<task_ptr> make_task(task::task_impl_ptr task_impl_ptr, task_info parent_d = task_info{});
+
+        // Must be called on target shard.
+        template<typename TaskImpl, typename... Args>
+        SEASTAR_CONCEPT( requires std::is_base_of_v<task::impl, TaskImpl> &&
```

Comment:
scylla supports c++20 so no need for `SEASTAR_CONCEPT` to encapsulate the `requires` concepts.